pkg/console/subresource/route/route.go

```diff
@@ -24,6 +30,49 @@ const (
 	defaultIngressController = "default"
 )
 
+// holds information about custom TLS certificate and its key
+type CustomTLSCert struct {
+	Certificate string
+	Key         string
+}
+
+func ApplyRoute(client routeclient.RoutesGetter, recorder events.Recorder, required *routev1.Route) (*routev1.Route, bool, error) {
+	existing, err := client.Routes(required.Namespace).Get(context.TODO(), required.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		requiredCopy := required.DeepCopy()
+		actual, err := client.Routes(requiredCopy.Namespace).
+			Create(context.TODO(), resourcemerge.WithCleanLabelsAndAnnotations(requiredCopy).(*routev1.Route), metav1.CreateOptions{})
+		// reportCreateEvent(recorder, requiredCopy, err)
+		return actual, true, err
+	}
+	if err != nil {
+		return nil, false, err
+	}
+	existingCopy := existing.DeepCopy()
+
+	if len(GetCanonicalHost(existingCopy)) == 0 {
+		return nil, false, customerrors.NewSyncError(fmt.Sprintf("route is not available at canonical host %s", existingCopy.Status.Ingress))
```

Comment:
Does this apply, even when using a custom route? `route is not available at canonical host %s`

Reply:
It should apply to custom route as well. The `routerName` should be set by the `ingressController` to default as well and the route needs to be admitted also

Reply:
this is no place for this check though, you should be able to apply the route no matter what its status is.

Also, are you 100% sure that this will be valid even for routes outside the cluster domain?